test/kitchen/test/integration/common/rspec/spec_helper.rb

```diff
@@ -449,6 +450,70 @@ def read_conf_file
   end
 end
 
+shared_examples_for 'an Agent with python3 enabled' do
+  it 'restarts after python_version is set to 3' do
+    conf_path = ""
+    if os != :windows
+      conf_path = "/etc/datadog-agent/datadog.yaml"
+    else
+      conf_path = "#{ENV['ProgramData']}\\Datadog\\datadog.yaml"
+    end
+    f = File.read(conf_path)
+    confYaml = YAML.load(f)
+    confYaml["python_version"] = 3
+    File.write(conf_path, confYaml.to_yaml)
+
+    output = restart
+    expect(output).to be_truthy
+  end
+
+  it 'runs Python 3 after python_version is set to 3' do
+    result = false
+    json_info_output = json_info
+    puts json_info_output
+    if json_info_output.key?('metadata') &&
+      json_info_output['metadata'].key?('systemStats') &&
+      json_info_output['metadata']['systemStats'].key?('pythonV')
+        pythonV = json_info_output['metadata']['systemStats']['pythonV']
```

Comment:
note: I don't know how many times you've tried to run the kitchen tests with this new test, but I think it may take some time for the Agent to be fully initialized (including having loaded the python interpreter) and actually expose `pythonV` in expvar.

So I think this test has chances of being a bit flaky if for some reason the Agent takes some time to initialize. Ideally there would be some logic here to wait until the expvar key is actually present (and timeout after x seconds).

No need to address this in this PR, but please be on the lookout for kitchen flakiness once this is merged, and think about this potential edge case if you see flakiness :)

Reply:
Actually I need to address this, I've tested 4 times, and each time it has failed at least once, so I'll add the timeout logic